### PR TITLE
Implement support for LocalDate and OffsetDateTime in RequestInformation

### LINKED ISSUE #1583: prepare for std uri template 2 and date time drop
related https://github.com/std-uritemplate/std-uritemplate/pull/245
The following types need to the normalized to their RFC3339 representation before they are passed to std URI template when they are present are query or path parameter values
Date, OffsetDateTime their collection representations (array, List)

### PR DESCRIPTION
**Pull request by espenvis**
Implements behavior for `LocalDate` and `OffsetDateTime` in RequestInformation's `getSanitizedValues`. Added accompanying test for `LocalDate` path parameter resolution.

Edit: scope is extended to include `LocalTime` and `PeriodAndDuration`.

closes #1583

### CODE REVIEW

**Comment by espenvis** [comment]
Marking as draft until we can evaluate extending this issue to tackle `time` and `duration` amongst other types.

**Comment by espenvis** [comment]
@microsoft-github-policy-service agree company="VisBook AS"

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java [hunk 1]
```diff
@@ -473,6 +474,10 @@ private static Object getSanitizedValues(Object value) {
             return ((ValuedEnum) value).getValue();
         } else if (value instanceof UUID) {
             return value.toString();
+        } else if (value instanceof OffsetDateTime) {
+            return ((OffsetDateTime) value).format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        } else if (value instanceof LocalDate) {
```
Comment (baywet) [inline components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java:479]: here we'd be missing LocalTime and PeriodAndDuration

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Could you also bump the dependency now that it's been released please?
(this way we can check there are no regressions)

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java (hunk 1 above)
Comment (espenvis) [reply components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java:479]: Yes, I will implement this now that it's been confirmed we want to support these.

**Comment by espenvis** [comment]
Updated to include `time` and `duration` datatype support (`LocalTime` and `PeriodAndDuration`). Also merged upstream changes. The version is now bumped up to `1.5.0` and all tests pass. Let me know if I've missed something.

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java (hunk 1 above)
Comment (espenvis) [reply components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java:479]: These are now implemented.

**Review by baywet — COMMENTED** [review]
Thank you for making the changes!
Can you also bump std uri template to v2 so we can confirm there's no end to end regression please?
https://github.com/microsoft/kiota-java/blob/bf66598153485fea2442dcf108d00e4ff6b8ab30/components/abstractions/gradle/dependencies.gradle#L11

**Review by baywet — DISMISSED** [review]
Thank you for making the changes!

**Comment by espenvis** [comment]
> Thank you for making the changes! Can you also bump std uri template to v2 so we can confirm there's no end to end regression please?
> 
> https://github.com/microsoft/kiota-java/blob/bf66598153485fea2442dcf108d00e4ff6b8ab30/components/abstractions/gradle/dependencies.gradle#L11

Yes. Apologies, I missed your earlier question.

It seems this causes an exception in `UriTemplateTest.java:32` - this passes parameters directly to `StdUriTemplate` and will fail.

**Comment by baywet** [comment]
@espenvis thanks for the information.
I think you can remove the last line in the CSV and the date parameter in the unit test. (make sure you pull beforehand since I've applied formatting to the csv file)

**Comment by espenvis** [comment]
> @espenvis thanks for the information. I think you can remove the last line in the CSV and the date parameter in the unit test. (make sure you pull beforehand since I've applied formatting to the csv file)

Yep, that's what I figured. I'll do that.

**Comment by espenvis** [comment]
Test still fails due to a JUnit error of some sort. I'll have a look.

**Comment by espenvis** [comment]
Seems it complained about the change in delimiter. Pushing a fix for that.

**Review by baywet — APPROVED** [review]
Thank you for making the changes!

**Review by baywet — APPROVED** [review]

**Pull request merged**